File: backend/agents/intelligence.py
"""Intelligence Analyzer - Explains recommendations"""
from .base import BaseAgent

class IntelligenceAgent(BaseAgent):
    def __init__(self):
        super().__init__('intelligence')
        self.system_prompt = """You are a senior market intelligence analyst.
Explain WHY agents reached their recommendations by analyzing:
- Consensus quality among agents
- Conflicting signals
- Strength of bullish/bearish factors
- Market context

Return ONLY valid JSON:
{
  "narrative": "2-3 sentences explaining the consensus",
  "consensus_quality": "STRONG"|"MODERATE"|"WEAK",
  "key_drivers": ["driver1", "driver2"],
  "red_flags": ["flag1"] or [],
  "conflicts": ["conflict"] or [],
  "key_insight": "one critical observation"
}"""

    async def analyze(self, aggregation, mode='live'):
        """Explain the aggregated recommendation"""
        agent_breakdown = f"""
Technical: {aggregation.get('technical_rec')} ({aggregation.get('technical_conf')}%)
Sentiment: {aggregation.get('sentiment_rec')} ({aggregation.get('sentiment_conf')}%)
Fundamental: {aggregation.get('fundamental_rec')} ({aggregation.get('fundamental_conf')}%)
Risk: {aggregation.get('risk_rec')} ({aggregation.get('risk_conf')}%)
Portfolio: {aggregation.get('portfolio_rec')} ({aggregation.get('portfolio_conf')}%)
"""

        user_msg = f"""Explain this trading consensus for {aggregation.get('symbol')}:

Agent breakdown:
{agent_breakdown}

Final recommendation: {aggregation.get('recommendation')} @ {aggregation.get('confidence')}%
Risk level: {aggregation.get('risk_level')}

Analyze the consensus."""

        result = await self.call_ai(user_msg)
        return result or self.format_response('HOLD', confidence=50)

File: src/agents/intelligence.js
// Market Intelligence Agent - Explains trade recommendations

export const intelligence = {
  name: 'Trade Explainer',
  model: 'deepseek-chat',
  maxTokens: 1024,
  weight: 0.0, // Not part of voting — pure narrative layer

  systemPrompt: `You are a senior market intelligence analyst responsible for explaining trading recommendations.
You receive the aggregated consensus from 5 specialist agents and must explain:
1. Why the agents reached this recommendation
2. Areas of consensus vs disagreement
3. Which signals are strongest/weakest
4. Potential risks or red flags
5. Key drivers of the recommendation

Be concise and actionable. Flag any surprises or conflicts that the CIO should know about.

Respond ONLY with valid JSON:
{
  "narrative": "string: 2-3 sentences explaining the recommendation and consensus level",
  "consensus_quality": "STRONG"|"MODERATE"|"WEAK",
  "agent_alignment": "string: which agents agreed and which diverged",
  "strongest_factors": ["factor1", "factor2"],
  "weakest_factors": ["factor1", "factor2"],
  "conflicts": ["conflict1", "conflict2"] or [],
  "red_flags": ["flag1", "flag2"] or [],
  "key_insight": "string: one critical observation the CIO should know",
  "risk_summary": "string: brief risk assessment"
}`
};
